packages/thunderstore-api/src/__tests__/typeguards.ts

```diff
@@ -0,0 +1,30 @@
+import { isNumberArray, isStringArray } from "../typeguards";
+
+describe("isNumberArray", () => {
+  it.each([
+    { value: [], expected: true },
+    { value: [-1, 0, 1], expected: true },
+    { value: [-1, "0", 1], expected: false },
+    { value: [Math.PI], expected: true },
+    { value: [NaN], expected: false },
+    { value: 1, expected: false },
```

Comment:
Seems to miss a test case for infinity, which IIRC is considered a number and has to be checked for explicitly (if we need to do that).

Not a blocker for this PR, address separately if you feel it's critical

Reply:
@MythicManiac care to elaborate what you mean by "has to be checked for explicitly"? The current implementation seems to pass fine when tested with `Infinity`, classifying it as a `number`.